src/containers/Validators/index.tsx

```diff
@@ -0,0 +1,229 @@
+import React, { useContext, useEffect, useState, useRef } from 'react'
+import axios from 'axios'
+import { useTranslation } from 'react-i18next'
+import { useQuery } from 'react-query'
+import NoMatch from '../NoMatch'
+import Loader from '../shared/components/Loader'
+import { Tabs } from '../shared/components/Tabs'
+import {
+  analytics,
+  ANALYTIC_TYPES,
+  FETCH_INTERVAL_VHS_MILLIS,
+  NOT_FOUND,
+  SERVER_ERROR,
+} from '../shared/utils'
+import { getLedger } from '../../rippled'
+import SimpleTab from './SimpleTab'
+import { HistoryTab } from './HistoryTab'
+import './validator.scss'
+import SocketContext from '../shared/SocketContext'
+
+const ERROR_MESSAGES = {
+  [NOT_FOUND]: {
+    title: 'validator_not_found',
+    hints: ['check_validator_key'],
+  },
+  default: {
+    title: 'generic_error',
+    hints: ['not_your_fault'],
+  },
+}
+
+const getErrorMessage = (error: keyof typeof ERROR_MESSAGES | null) =>
+  (error && ERROR_MESSAGES[error]) || ERROR_MESSAGES.default
+
+interface Props {
+  match: {
+    path?: string
+    params: {
+      identifier?: string
+      tab?: string
+    }
+  }
+}
+
+interface ValidatorData {
+  domain?: string
+  // eslint-disable-next-line camelcase -- from VHS
+  master_key?: string
+  // eslint-disable-next-line camelcase -- from VHS
+  signing_key?: string
+}
+
+const Validator = (props: Props) => {
+  const [reports, setReports] = useState([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [error, setError] = useState(null)
+  const [data, setData] = useState<ValidatorData>({})
+  const [width, setWidth] = useState(0)
+  const ref = useRef<any>(null)
+  const { t, i18n } = useTranslation()
```

Comment:
`useLanguage` hook will give you the language.  `i18n` is the i18n-next object and is not the language string.

Reply:
There's no `useLanguage` hook in `react-i18next`

Reply:
Its in the explorer.  I wrote it to avoid having to access `i18n.resolvedLanguage`